src/corredor/helpers/messaging.ts

```diff
@@ -61,13 +78,18 @@ export class Messaging {
    * @param filter
    * @returns {Promise<{filter: Object, set: Channel[]}>}
    */
-  async findChannels (filter) {
+  async findChannels (filter: string|StrIndex<string>): Promise<StrIndex<unknown>> {
```

Comment:
I suggest you replace this generic approach with specialised interfaces.
For example, you should replace filter: `string| filter: string|StrIndex<string>` with `filter: string|ChannelFilter` where ChannelFilter is an interface with all supported fields.

Same goes for returned value. In this example it could be easily defined as:

```
interface ChannelResult {
  filter: ChannelFilter
  set: Array<Channel>
}
```

Reply:
And this is, of corse, applicable to all other case and uses of StrIndex